lib/kotkowo_web/live/adoption_live/looking_for_new_home.ex

```diff
@@ -2,37 +2,385 @@ defmodule KotkowoWeb.AdoptionLive.LookingForNewHome do
   @moduledoc false
   use KotkowoWeb, :live_view
 
+  import KotkowoWeb.Components.Drawers
+  import KotkowoWeb.Components.Static.HowYouCanHelpSection
+
+  alias Kotkowo.Attributes.Color
+  alias Kotkowo.Attributes.Seniority
+  alias Kotkowo.Attributes.Sex
+  alias Kotkowo.GalleryImage
   alias Kotkowo.StrapiClient
 
+  @first_page 1
+  @default_limit "30"
   @impl true
   def mount(_params, _session, socket) do
-    {:ok, cats} = StrapiClient.list_cats()
+    {:ok, cats} = StrapiClient.list_cats(false, false)
 
     socket =
-      socket
-      |> assign(:cats, cats)
-      |> assign(:filtered_cats, cats)
+      assign(socket, :cats, cats)
 
     {:ok, socket}
   end
 
   @impl true
-  def handle_event("cat_search", %{"cat_search" => search}, socket) do
-    cats = socket.assigns.cats
+  def handle_params(params, _uri, socket) do
+    limit = params |> Map.get("limit", @default_limit) |> parse_limit()
+    page = params |> Map.get("page", Integer.to_string(@first_page)) |> parse_limit()
+
+    name = params |> Map.get("name", "") |> String.downcase()
+    seniority = Map.get(params, "seniority", [])
+    sexes = Map.get(params, "sexes", [])
+    castrated = params |> Map.get("castrated", nil) |> parse_checkbox()
+    tag = params |> Map.get("tag", "") |> String.downcase()
+    colors = Map.get(params, "colors", [])
+    date_to = Map.get(params, "date_to", "")
+    date_from = Map.get(params, "date_from", "")
```

Comment:
clean this up by defining this module attribute:
```elixir
@params_default %{
  limit: 30,
  page: 1,
  name: "",
  seniority: [],
  sexes: [],
  castrated: nil,
  tag: "",
  colors: [],
  date_to: "",
  date_from: ""
}

@params_keys @params_default |> Map.keys() |> Enum.map(&to_string/1)
```

Add a function that would parse all the params at once:

```elixir
defp parse_param({"page", value}) do
  page = :placeholder # parse the value into this
  {:page, page}
end
# for things that don't require more parsing
defp parse_param({key, value}) do
  {String.to_existing_atom(key), value}
end
```

then you can:
```elixir
parsed_params = Map.merge(
  @params_default,
  Enum.into(params, %{}, &parse_param/1)
)
```

And then later you access using the `.`(dot)

```elixir
    filtered_cats =
      socket.assigns.cats
      |> search_tags(parsed_params.tag) # <- dot used here
```